src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala

```diff
@@ -201,23 +201,17 @@ class EntityService(rawlsDAO: RawlsDAO, importServiceDAO: ImportServiceDAO, goog
     // The async path only supports upsert TSVs, not update TSVs. Import Service only operates in upsert mode;
     // if we tried to send an update TSV through it, it would lose its "update-only" restriction and become an upsert.
     if (isAsync) {
-      if (!isUpsert) {
-        // until Rawls & Import Service implement update-only TSVs, throw a BadRequest if the user requested async
-        // but sent an update TSV.
-        Future(RequestCompleteWithErrorReport(BadRequest, "Update-only TSVs cannot use async mode"))
-      } else {
-        asyncUpsert(workspaceNamespace, workspaceName, rawlsCalls, userInfo).recover {
-          case e: Exception =>
-            RequestCompleteWithErrorReport(InternalServerError, "Unexpected error during async TSV import", e)
-        }
+      asyncUpsert(workspaceNamespace, workspaceName, isUpsert, rawlsCalls, userInfo).recover {
+        case e: Exception =>
+          RequestCompleteWithErrorReport(InternalServerError, "Unexpected error during async TSV import", e)
       }
     } else {
       val rawlsResponse = rawlsDAO.batchUpsertEntities(workspaceNamespace, workspaceName, entityType, rawlsCalls)(userInfo)
       handleBatchRawlsResponse(entityType, rawlsResponse)
     }
   }
 
-  private def asyncUpsert(workspaceNamespace: String, workspaceName: String,
+  private def asyncUpsert(workspaceNamespace: String, workspaceName: String, isUpsert: Boolean,
```

Comment:
Seeing a method called `asyncUpsert` with a parameter called `isUpsert` is jarring to me.  It's called `asyncUpsert` so  `isUpsert` should always be true at least for the easiest comprehension.  I understand that the isUpsert name is baked in to Rawls and ImportService already so doing something like changing the param to `isUpdateOnly` and inverting the logic here in Orch will make the code less consistent with Rawls and ImportService.  What if we renamed this method to `asyncUpdateOrInsert` or just just `asyncUpdate`?  I'll leave it to you to do what you think is best.

Reply:
no, that's good feedback, and my fault - I had that method name left over from when it truly was upsert-only, and failed to rename it!

Also, wow, your comment called out another problem in the synchronous path, which I am fixing too.

Reply:
Cool, good catch. High five! :hand: